### PR TITLE
TaxCategory Type Zone integration test refactor

### PR DESCRIPTION
**Pull request by barbara79**
1. code and add tests that cover the created code. Your code should be warning-free.
   * [ ] tests existing
1. stick to PSR-2 and and don't reformat existing code.
   * [ ] code style check succesful

### CODE REVIEW

File: tests/integration/Zone/ZoneQueryRequestTest.php [hunk 1]
```diff
@@ -3,104 +3,81 @@
...
+
+                $this->assertInstanceOf(Zone::class, $zone);
+                $this->assertSame($zone->getId(), $result->getId());
+                $this->assertSame($zone->getKey(), $result->getKey());
+            }
+        );
     }
-
+    //TODO for Consistancy: should I change the place of this test? in other part it's in createRequestTest or in Update
     public function testDeleteByKey()
```
Comment (barbara79) [inline tests/integration/Zone/ZoneQueryRequestTest.php]: for consistancy: should we change the place of this test? in other part of the code it's in createRequestTest or in Update

File: tests/integration/TaxCategory/TaxCategoryUpdateRequestTest.php [hunk 2]
```diff
@@ -3,213 +3,193 @@
...
-        $this->assertNotSame($taxCategory->getVersion(), $result->getVersion());
+        $client = $this->getApiClient();
+
+        TaxCategoryFixture::withUpdateableDraftTaxCategory(
+            $client,
+            function (TaxCategoryDraft $draft) {
+                return $draft->setDescription('set-description');
+            },
+            function (TaxCategory $taxCategory) use ($client) {
+                $description = 'new-description';
```
Comment (barbara79) [inline tests/integration/TaxCategory/TaxCategoryUpdateRequestTest.php]: and if we replace the test variable with something like ?
$description = Uuid::uuidv4() . 'new-description';

Comment (jenschude) [reply tests/integration/TaxCategory/TaxCategoryUpdateRequestTest.php]: Maybe better with `'new-description-' . Uuid::uuidv4()`

File: tests/integration/Zone/ZoneQueryRequestTest.php (hunk 1 above)
Comment (jenschude) [reply tests/integration/Zone/ZoneQueryRequestTest.php]: Not really necessary to move it.

File: tests/integration/TaxCategory/TaxCategoryUpdateRequestTest.php (hunk 2 above)
Comment (barbara79) [reply tests/integration/TaxCategory/TaxCategoryUpdateRequestTest.php]: as you proposed in the other PRs, I'm replacing with 
 'new-description-' . TaxCategoryFixture::uniqueTaxCategoryString();

**Pull request merged**